src/core/services/compiler/compiler.js

```diff
@@ -10,74 +10,90 @@ angular
 
 /**
  * @ngdoc service
- * @name $mdCompiler
+ * @name $mdCompilerProvider
  * @module material.core.compiler
  * @description
- * The $mdCompiler service is an abstraction of AngularJS's compiler, that allows developers
- * to easily compile an element with options like in a Directive Definition Object.
+ * The `$mdCompiler` is able to respect the AngularJS `$compiler.preAssignBindingsEnabled` state when using
+ * AngularJS versions between 1.5.10 and 1.7.0.
+ * See the [AngularJS documentation for `$compile.preAssignBindingsEnabled`
+ * ](https://code.angularjs.org/1.6.10/docs/api/ng/provider/$compileProvider#preAssignBindingsEnabled)
+ * for more information.
  *
- * > The compiler powers a lot of components inside of AngularJS Material.
- * > Like the `$mdPanel` or `$mdDialog`.
+ * To enable/disable whether the controllers of dynamic AngularJS Material components
+ * (i.e. dialog, panel, toast, bottomsheet) respect the AngularJS `$compile.preAssignBindingsEnabled` flag,
+ * call the AngularJS Material method: `$mdCompilerProvider.respectPreAssignBindingsEnabled(boolean)`.
+ *
+ * This AngularJS Material *flag* doesn't affect directives/components created via regular AngularJS methods.
+ * These constitute the majority of AngularJS Material and user-created components.
+ *
+ * Only dynamic construction of elements such as Dialogs, Panels, Toasts, BottomSheets, etc. may be affected.
+ * Invoking `$mdCompilerProvider.respectPreAssignBindingsEnabled(true)` will make **bindings** in
+ * AngularJS Material custom components like `$mdDialog`, `$mdPanel`, `$mdToast`, or `$mdBottomSheet`
+ * only available in the controller constructors.
```

Comment:
I believe this is wrong. (As mentioned below, invoking `$mdCompilerProvider.respectPreAssignBindingsEnabled(true)` can result in different behaviors, depending on `$compileProvider.preAssignBindingsEnabled()` and AngularJS version.)

Reply:
I agree, I've cleaned this up.